Baby Monitor/Source Files/AppDelegate.swift

```diff
@@ -13,12 +13,21 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
     
     var window: UIWindow?
     var rootCoordinator: RootCoordinatorProtocol?
-    let appDependencies = AppDependencies()
+    private(set) var appDependencies = AppDependencies()
     
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
         window = UIWindow(frame: UIScreen.main.bounds)
         rootCoordinator = RootCoordinator(window!, appDependencies: appDependencies)
         rootCoordinator?.start()
+        rootCoordinator?.onResetApp = { [weak self] in
+            // This line is extremely important. After resseting the app we may want to establish new
+            // WebRTC connection. Thanks to deinitializing and initializing AppDependencies again we are
+            // sure that old connection is properly cleared.
+            // UPD: TODO: Needed to check whether other view models are not keeping reference to the old dependencies.
+            let dependencies = AppDependencies()
+            self?.rootCoordinator?.update(dependencies: dependencies)
+            self?.appDependencies = dependencies
```

Comment:
This is potentially very dangerous an leaky, but, for now, there is no alternative.

Reply:
😔